client/lib/posts/post-edit-store.js

```diff
@@ -99,7 +99,7 @@ function updatePost( post ) {
 	} );
 }
 
-function initializeNewPost( siteId, options ) {
+function initializeNewPost( siteId, options, site ) {
```

Comment:
We pass both `site` and `siteId`. Maybe we can lose the latter? 🙂 

See: https://github.com/Automattic/wp-calypso/pull/16017/files#r151115432